R/errorChecks.r

```diff
@@ -1,7 +1,8 @@
 #' @importFrom dplyr distinct
 
-errorChecks <- function(taxa = NULL, site = NULL, time_period = NULL, startDate = NULL,
-                        endDate = NULL, Date = NULL, time_periodsDF = NULL, dist = NULL, sim = NULL,
+errorChecks <- function(taxa = NULL, site = NULL, survey = NULL, closure_period = NULL, time_period = NULL, 
```

Comment:
@drnickisaac The error you get looks like this:

```
── 2. Failure: Test errors and warnings - siteSelection (@testsiteSelection.R#82
`... <- NULL` did not throw an error.
```

Originates with this change. You have changed the third argument to be for survey, not for time period. This means that all the checks that are written assuming that time_period is the third argument will no longer work as expected, for example

```r
errorChecks(taxa, site, time_period)
```

which appears in siteSelection.r L50.

The easiest way to fix this would be to place survey at the end of the current argument list to `errorChecks`. 

You shouldnt need to change anything else since you specify survey in your call to errorChecks.

`formatOccDet` L98
```r
errorChecks(taxa = taxa, site = site, survey = survey, closure_period = closure_period)
```

Once you place survey at the end of the current argument list to `errorChecks` (errorChecks.r L3) this should get rid of the error. I will open a new issue for me to review all calls to errorChecks and ensure that they are not dependent on argument order so that this does not happen in the future

Reply:
Addition! We need to move both survey and closure_period